pallets/utility/src/lib.rs

```diff
@@ -618,184 +562,13 @@ pub mod pallet {
-            )?);
-            Ok(())
-        }
-
-        /// Dispatch multiple calls from the sender's origin.
-        ///
-        /// This will execute all calls, in order, irrespective of failures.
-        /// Any failures will be available in a `BatchOptimisticFailed` event.
-        ///
-        /// May be called from root or a signed origin.
-        ///
-        ///# Parameters
-        /// - `calls`: The calls to be dispatched from the same origin.
-        ///
-        ///
-        /// # Weight
-        /// - The sum of the weights of the `calls`.
-        /// - One event.
-        ///
-        /// This will return `Ok` in all circumstances except an unsigned origin.
-        /// To determine the success of the batch, an event is deposited.
-        /// If any call failed, then `BatchOptimisticFailed` is deposited,
-        /// with a vector of event counts for each call as well as a vector
-        /// of errors.
-        /// If all were successful, then the `BatchCompletedOld` event is deposited.
-        ///
-        /// POLYMESH: deprecated.
-        #[pallet::call_index(8)]
-        #[pallet::weight({
-                let dispatch_infos = calls.iter().map(|call| call.get_dispatch_info()).collect::<Vec<_>>();
-                let dispatch_weight = dispatch_infos.iter()
-                    .map(|di| di.weight)
-                    .fold(Weight::zero(), |total: Weight, weight: Weight| total.saturating_add(weight))
-                    .saturating_add(<T as Config>::WeightInfo::batch_optimistic(calls.len() as u32));
-                let dispatch_class = {
-                    let all_operational = dispatch_infos.iter()
-                        .map(|di| di.class)
-                        .all(|class| class == DispatchClass::Operational);
-                    if all_operational {
-                        DispatchClass::Operational
-                    } else {
-                        DispatchClass::Normal
-                    }
-                };
-                (dispatch_weight, dispatch_class)
-            })]
-        #[allow(deprecated)]
-        #[deprecated(note = "Please use `force_batch` instead.")]
-        pub fn batch_optimistic(
-            origin: OriginFor<T>,
-            calls: Vec<<T as Config>::RuntimeCall>,
-        ) -> DispatchResult {
-            let is_root = Self::ensure_root_or_signed(origin.clone())?;
-
-            // Optimistically (hey, it's in the function name, :wink:) assume no errors.
-            Self::deposit_event(Self::run_batch(origin, is_root, calls, false));
-            Ok(())
-        }
-
         /// Send a call through an indexed pseudonym of the sender.
         ///
         /// Filter from origin are passed along. The call will be dispatched with an origin which
         /// use the same filter as the origin of this call.
         ///
         /// The dispatch origin for this call must be _Signed_.
-        #[pallet::call_index(9)]
+        #[pallet::call_index(6)]
```

Comment:
We should leave the extrinsic call_index values fixed.  Any change to these will require a change to the `transaction_version` and update to the Ledger app.